grails-app/assets/javascripts/projectActivityPlan.js

```diff
@@ -307,7 +307,11 @@ var PlanStage = function (stage, activities, planViewModel, isCurrentStage, proj
                 self.updateStageStatus(url, this.reason(), this.rejectionCategory());
             }
         };
-        ko.applyBindings(reasonViewModel, $reasonModal[0]);
+
+        var hasDataBinding = !!ko.dataFor($reasonModal[0]);
+        if (!hasDataBinding) {
+            ko.applyBindings(reasonViewModel, $reasonModal[0]);
+        }
         $reasonModal.modal({backdrop: 'static', keyboard:true, show:true}).on('hidden', function() {ko.cleanNode($reasonModal[0])});
```

Comment:
The actual fix here is to listen to the correct event - the problem here is the models in bs4 fire 'hidden.bs.modal', not 'hidden'.  (See equivalent in reporting.js).
There are two places in this file with this bug, can you fix the other one while you are fixing this?  Thanks.